### PR TITLE
fix(applications-service-api): ni interested party db encoding error handling

### PR DESCRIPTION
**Pull request by JamesP-KC**
## Describe your changes

ASB-2080

Add some additional error handling for the scenario where inserting a row into the NI database for a table/column which does not have utf8 character encoding enabled.

Also added a unit test for the interested party repository.

## Useful information to review or test

Sequelize throws a specific error for this scenario (`ER_TRUNCATED_WRONG_VALUE_FOR_FIELD`)

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+			expect(mockInterestedPartyCreate).toHaveBeenCalledWith(input);
+			expect(result).toEqual({
+				id: 1234,
+				...input
+			});
+		});
+
+		describe('sequelize throws error', () => {
+			it("throws apierror if sequelize 'wrong value' error is thrown", async () => {
```
Comment (Yakitrak) [inline packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js:35]: Nitpick so feel free to ignore but would good to do it in BDT format:

```javascript
describe('when submitting invalid characters', () => {
			it("should throw 'wrong value' api error", async () => {
```

**Review by Yakitrak — APPROVED** [review]
Minor comment on test but LGTM - approved

File: packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js (hunk 1 above)
Comment (JamesP-KC) [reply packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js]: Sounds good to me, have updated

File: packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+				await expect(() => createInterestedParty({})).rejects.toEqual({
+					code: 400,
+					message: {
+						errors: [errorMessage]
+					}
+				});
+			});
+
+			it('should rethrow other errors', async () => {
```
Comment (Yakitrak) [inline packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js:57]: This likely belongs in a different describe block.

File: packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+				mockInterestedPartyCreate.mockImplementationOnce(() => {
+					throw someError;
+				});
+
+				await expect(() => createInterestedParty({})).rejects.toEqual(someError);
+			});
+		});
+	});
+
+	describe('updateInterestedParty', () => {});
```
Comment (JamesP-KC) [inline packages/applications-service-api/__tests__/unit/repositories/interestedParty.repository.test.js:71]: forgot to add test

**Pull request merged**